scipy/_build_utils/_fortran.py

```diff
@@ -67,3 +73,288 @@ def gfortran_legacy_flag_hook(cmd, ext):
+            new_sources = _generic_patch_sources(build_info['sources'], patch_source_func,
+                                                 source_fnpart)
+            build_info['sources'][:] = new_sources
+        else:
+            ext.depends.extend(ext.sources)
+            new_sources = _generic_patch_sources(ext.sources, patch_source_func,
+                                                 source_fnpart)
+            ext.sources[:] = new_sources
+
+
+def _generic_patch_sources(filenames, patch_source_func, source_fnpart, root_dir=None):
+    """
+    Patch Fortran sources, creating new source files.
+
+    Parameters
+    ----------
+    filenames : list
+        List of Fortran source files to patch.
+        Files not ending in ``.f`` or ``.f90`` are left unaltered.
+    patch_source_func : callable(filename, old_contents) -> new_contents
+        Function to apply to file contents, returning new file contents
+        as a string.
+    source_fnpart : str
+        String to append to the modified file basename before extension.
+    root_dir : str, optional
+        Source root directory. Default: cwd
+
+    Returns
+    -------
+    new_filenames : list
+        List of names of the newly created patched sources.
+
+    """
+    new_filenames = []
+
+    if root_dir is None:
+        root_dir = os.getcwd()
+
+    root_dir = os.path.abspath(root_dir)
+    src_dir = os.path.join(root_dir, _get_build_src_dir())
+
+    for src in filenames:
+        base, ext = os.path.splitext(os.path.basename(src))
+
+        if ext not in ('.f', '.f90'):
+            new_filenames.append(src)
+            continue
+
+        with open(src, 'r') as fsrc:
+            text = patch_source_func(src, fsrc.read())
+
+        # Generate useful target directory name under src_dir
+        src_path = os.path.abspath(os.path.dirname(src))
+
+        for basedir in [src_dir, root_dir]:
+            if os.path.commonpath([src_path, basedir]) == basedir:
+                rel_path = os.path.relpath(src_path, basedir)
+                break
+        else:
+            raise ValueError(f"{src!r} not under {root_dir!r}")
+
+        dst = os.path.join(src_dir, rel_path, base + source_fnpart + ext)
+        write_file_content(dst, text)
+
+        new_filenames.append(dst)
+
+    return new_filenames
+
+
+def write_file_content(filename, content):
+    """
+    Write content to file, but only if it differs from the current one.
+    """
+    if os.path.isfile(filename):
+        with open(filename, 'r') as f:
+            old_content = f.read()
+
+        if old_content == content:
+            return
+
+    dirname = os.path.dirname(filename)
+    if not os.path.isdir(dirname):
+        os.makedirs(dirname)
+
+    with open(filename, 'w') as f:
+        f.write(content)
+
+
+def get_blas_lapack_symbols():
+    cached = getattr(get_blas_lapack_symbols, 'cached', None)
+    if cached is not None:
+        return cached
+
+    # Symbols not in Cython Lapack interface
```

Comment:
This probably changes for newer LAPACK versions, so add version to this comment?

Reply:
Changed it to load these directly from the generator script, to avoid duplicating the list here.